lte/gateway/python/magma/subscriberdb/client.py

```diff
@@ -77,32 +78,47 @@ def __init__(
         self._grpc_client_manager = grpc_client_manager
 
     async def _run(self) -> None:
-        subscribers = await self._get_subscribers()
-        if subscribers is not None:
-            self._process_subscribers(subscribers)
+        subscribers, no_updates = await self._get_subscribers()
+        # No need to update local db if it's already up-to-date.
+        if subscribers is None or no_updates:
+            return
+        self._process_subscribers(subscribers)
 
     async def _get_subscribers(self) -> [SubscriberData]:
         subscriberdb_cloud_client = self._grpc_client_manager.get_client()
         subscribers = []
         req_page_token = ""  # noqa: S105
         res_page_token = "start_token"  # noqa: S105
+        cur_digest = SubscribersDigest(
+            md5_base64_digest=self._store.get_current_digest(),
+        )
+        no_updates = False
         sync_start = datetime.datetime.now()
-        while res_page_token != "":  # noqa: S105
+
+        # If response.no_updates = True, it means local data is up-to-date &
+        # therefore no need for updates from the cloud.
+        while res_page_token != "" and not no_updates:  # noqa: S105
             try:
                 req = ListSubscribersRequest(
                     page_size=self._subscriber_page_size,
                     page_token=req_page_token,
+                    digest=cur_digest,
                 )
-                response = await grpc_async_wrapper(
+                res = await grpc_async_wrapper(
                     subscriberdb_cloud_client.ListSubscribers.future(
                         req,
                         self.SUBSCRIBERDB_REQUEST_TIMEOUT,
                     ),
                     self._loop,
                 )
-                subscribers.extend(response.subscribers)
-                res_page_token = response.next_page_token
-                req_page_token = response.next_page_token
+                subscribers.extend(res.subscribers)
+                res_page_token = res.next_page_token
+                req_page_token = res.next_page_token
+                no_updates = res.no_updates
+
+                if cur_digest != res.digest:
+                    cur_digest = res.digest
```

Comment:
This if-guard doesn't do anything, can just set it
```suggestion
                cur_digest = res.digest
```